test-programs/account-compression-test/tests/merkle_tree_tests.rs

```diff
@@ -32,6 +34,311 @@ use solana_sdk::{
+    let mut account_share_data = AccountSharedData::from(account);
+    account_share_data.set_lamports(lamports);
+    context.set_account(nullifier_queue_pubkey, &account_share_data);
+    let account = context
+        .banks_client
+        .get_account(*nullifier_queue_pubkey)
+        .await
+        .unwrap()
+        .unwrap();
+    let mut data = account.data.clone();
+    let nullifier_queue =
+        &mut unsafe { nullifier_queue_from_bytes_zero_copy_mut(&mut data).unwrap() };
+    for i in current_index..(capacity) {
+        let array_element = nullifier_queue.by_value_index(i, None).unwrap();
+        assert_eq!(array_element.value_biguint(), (i).to_biguint().unwrap());
+    }
+}
+
+fn find_overlapping_probe_index(
+    initial_value: usize,
+    start_replacement_value: usize,
+    capacity_values: usize,
+) -> usize {
+    for salt in 0..10000 {
+        let replacement_value = start_replacement_value + salt;
+
+        for i in 0..20 {
+            let probe_index = (initial_value.clone()
+                + i.to_biguint().unwrap() * i.to_biguint().unwrap())
+                % capacity_values.to_biguint().unwrap();
+            let replacement_probe_index = (replacement_value.clone()
+                + i.to_biguint().unwrap() * i.to_biguint().unwrap())
+                % capacity_values.to_biguint().unwrap();
+            if probe_index == replacement_probe_index {
+                return replacement_value;
+            }
+        }
+    }
+    panic!("No value with overlapping probe index found!");
+}
+async fn fail_insert_into_full_queue(
+    context: &mut ProgramTestContext,
+    nullifier_queue_pubkey: &Pubkey,
+    merkle_tree_pubkey: &Pubkey,
+    elements: Vec<[u8; 32]>,
+) {
+    let payer = context.payer.insecure_clone();
+
+    let err = insert_into_nullifier_queues(
+        &elements,
+        &payer,
+        &payer,
+        nullifier_queue_pubkey,
+        merkle_tree_pubkey,
+        context,
+    )
+    .await;
+    assert_eq!(
+        err.map_err(|e| e.to_string()),
+        Err(BanksClientError::TransactionError(
+            solana_sdk::transaction::TransactionError::InstructionError(
+                0,
+                InstructionError::Custom(HashSetError::Full.into())
+            )
+        ))
+        .map_err(|e| e.to_string()),
+    );
+}
+
+pub async fn set_state_merkle_tree_sequence(
+    context: &mut ProgramTestContext,
+    merkle_tree_pubkey: &Pubkey,
+    sequence_number: u64,
+    lamports: u64,
+) {
+    println!("advance to sequence number {}", sequence_number);
+    // is in range 9 - 10 in concurrent mt
+    // offset for sequence number
+    let offset_start = 6 * 8 + 8 + 4 * 32 + 8 * 9;
+    let offset_end = offset_start + 8;
+    let mut merkle_tree = context
+        .banks_client
+        .get_account(*merkle_tree_pubkey)
+        .await
+        .unwrap()
+        .unwrap();
+    merkle_tree.data[offset_start..offset_end].copy_from_slice(&sequence_number.to_le_bytes());
```

Comment:
And this could become

```suggestion
    merkle_tree.data[span_of!(ConcurrentMerkleTree26<Poseidon>, sequence_number)].copy_from_slice(&sequence_number.to_le_bytes());
```